addon/components/utils/address-form.ts

```diff
@@ -30,13 +29,52 @@ export default class extends Component<UtilsAddressFormArgs> {
   validPhoneNumber: boolean = true;
   countries = countries;
 
+  get useGoogleAutocomplete(): boolean {
+    return this.args.useGoogleAutocomplete ?? true;
+  }
+
+  @action
+  onSetup() {
+    if (isTesting()) return;
+    const loader = new Loader({
+      apiKey: getOwner(this).resolveRegistration('config:environment').google_map_api_key,
+      version: 'weekly'
+    });
+
+    loader
+      .importLibrary('places')
+      .then(({ Autocomplete }) => {
+        const input = document.querySelector('[data-control-name="address-form-address1"] input') as HTMLInputElement;
+        const options = {
+          fields: ['address_components'],
+          strictBounds: false,
+          types: ['address']
+        };
+
+        const autocomplete = new Autocomplete(input, options);
+
+        autocomplete.addListener('place_changed', () => {
+          const place = autocomplete.getPlace();
+          this.fillInAddress(place);
+        });
+        input.addEventListener('focusout', (event) => {
+          if ((<HTMLInputElement>event.target).value !== this.args.address.address1) {
+            (<HTMLInputElement>event.target).value = this.args.address.address1;
+          }
+        });
+      })
```

Comment:
nit: 
rename `onSetup` to `initAutoCompletion`
and move all this (line 47 to 65) to private method, `initInputListeners` maybe ?